lib/blacklight/search_helper.rb

```diff
@@ -286,22 +255,59 @@ def blacklight_solr
 
   private
 
-  ##
-  # @deprecated
-  def default_solr_doc_params(id=nil)
-    id ||= params[:id]
+    ##
+    # Retrieve a set of documents by id
+    # @overload fetch_many(ids, extra_controller_params)
+    # @overload fetch_many(ids, user_params, extra_controller_params)
+    def fetch_many(ids=[], *args)
+      if args.length == 1
+        Deprecation.warn(SearchHelper, "fetch_many with 2 arguments is deprecated")
+        user_params = params
+        extra_controller_params = args.first || {}
+      else
+        user_params, extra_controller_params = args
+        user_params ||= params
+        extra_controller_params ||= {}
+      end
+
+      query = search_builder.with(user_params).query(extra_controller_params.merge(solr_document_ids_params(ids)))
+      solr_response = repository.search(query)
+
+      [solr_response, solr_response.documents]
+    end
 
-    # add our document id to the document_unique_id_param query parameter
-    p = blacklight_config.default_document_solr_params.merge({
-      # this assumes the request handler will map the unique id param
-      # to the unique key field using either solr local params, the
-      # real-time get handler, etc.
-      blacklight_config.document_unique_id_param => id
-    })
+    alias_method :get_solr_response_for_document_ids, :fetch_many
```

Comment:
SearchHelper is usually included into the class that would call `get_solr_response_for_document_ids`, so yes.